include/coloring/pgr_edgeColoring.hpp

```diff
@@ -1,95 +1,162 @@
+#include <boost/graph/properties.hpp>
+#include <boost/config.hpp>
 
+#include <limits>
 #include <iostream>
 #include <algorithm>
 #include <vector>
+#include <map>
 
 #include "cpp_common/pgr_base_graph.hpp"
-#include "cpp_common/pgr_messages.h"
 #include "cpp_common/interruption.h"
 
 
+/** @file pgr_edgeColoring.hpp
+ * @brief The main file which calls the respective boost function.
+ *
+ * Contains actual implementation of the function and the calling
+ * of the respective boost function.
+ */
+
+//will adjust it later;
+using namespace boost;
+using namespace std;
+
 namespace pgrouting {
 namespace functions {
 
-template<class G>
-class Pgr_Bipartite : public pgrouting::Pgr_messages {
- public:
-            typedef typename G::V_i V_i;
-            std::vector<pgr_vertex_color_rt> print_Bipartite(
-                    G &graph) {
-                std::vector<pgr_vertex_color_rt> results;
-                std::vector <boost::default_color_type> partition(graph.num_vertices());
-                auto partition_map =
-                    make_iterator_property_map(partition.begin(), boost::get(boost::vertex_index, graph.graph));
-
-               /* abort in case of an interruption occurs (e.g. the query is being cancelled) */
-                CHECK_FOR_INTERRUPTS();
-                try {
-                boost::is_bipartite(graph.graph, boost::get(boost::vertex_index, graph.graph), partition_map);
-                     } catch (boost::exception const& ex) {
-                          (void)ex;
-                             throw;
-                     } catch (std::exception &e) {
-                         (void)e;
-                         throw;
-                      } catch (...) {
-                          throw;
-                      }
-                V_i v, vend;
-                for (boost::tie(v, vend) = vertices(graph.graph); v != vend; ++v) {
-                    int64_t vid = graph[*v].id;
-                    boost::get(partition_map, *v) ==
-                        boost::color_traits <boost::default_color_type>::white() ?
-                        results.push_back({vid, 0}) :results.push_back({vid, 1});
-                }
-                return results;
-            }
-            std::vector<pgr_vertex_color_rt> pgr_bipartite(
-                    G &graph ){
-                std::vector<pgr_vertex_color_rt> results;
-                bool bipartite = boost::is_bipartite(graph.graph);
-                if (bipartite) results = print_Bipartite(graph);
-                return results;
-            }
+//*************************************************************
+
+template < class G >
+class Pgr_edgeColoring {
+public:
+    typedef typename G::V V;
+    typedef typename G::E E;
+
+    //#new
+    typedef adjacency_list< vecS, vecS, undirectedS, no_property, size_t,
+            no_property >
+            Graph;
+
+    // typedef boost::adjacency_list < boost::listS, boost::vecS, boost::undirectedS > Graph;
```

Comment:
Wrap code you dont want the compiler to compile as follows. Never use comments.
```
#if 0
typedef boost::adjacency_list < boost::listS, boost::vecS, boost::undirectedS > Graph;
#endif
````